### ISSUE TITLE
Unable to load symbols, unable to save (.dump) the core dump.

### ISSUE THREAD

**Issue by MarkKharitonov** [microsoft/WinDbg-Feedback#120]
I have a .NET core dump taken on a Linux machine.

When I open it in WinDbg I am greeted with "64-bit machine not using 64-bit API":
```

************* Preparing the environment for Debugger Extensions Gallery repositories **************
   ExtensionRepository : Implicit
   UseExperimentalFeatureForNugetShare : true
   AllowNugetExeUpdate : true
   AllowNugetMSCredentialProviderInstall : true
   AllowParallelInitializationOfLocalRepositories : true

   -- Configuring repositories
      ----> Repository : LocalInstalled, Enabled: true
      ----> Repository : UserExtensions, Enabled: true

>>>>>>>>>>>>> Preparing the environment for Debugger Extensions Gallery repositories completed, duration 0.000 seconds

************* Waiting for Debugger Extensions Gallery to Initialize **************

>>>>>>>>>>>>> Waiting for Debugger Extensions Gallery to Initialize completed, duration 0.031 seconds
   ----> Repository : UserExtensions, Enabled: true, Packages count: 0
   ----> Repository : LocalInstalled, Enabled: true, Packages count: 36

Microsoft (R) Windows Debugger Version 10.0.25921.1001 AMD64
Copyright (c) Microsoft Corporation. All rights reserved.


Loading Dump File [C:\Temp\dumps\clean_coredump]
64-bit machine not using 64-bit API

************* Path validation summary **************
Response                         Time (ms)     Location
Deferred                                       srv*\\symbolsrv.xyz.com\symbols*\\symbolsrv.xyz.com\symbols\xyz*http://msdl.microsoft.com/download/symbols
Symbol search path is: srv*\\symbolsrv.xyz.com\symbols*\\symbolsrv.xyz.com\symbols\xyz*http://msdl.microsoft.com/download/symbols
Executable search path is: 
Generic Unix Version 0 UP Free x64
System Uptime: not available
Process Uptime: not available
................................................................
................................................................
................................................................
................................................................
................................................................
................................................................
................................................................
..........................................
*** WARNING: Unable to verify timestamp for libpthread-2.31.so
*** WARNING: Unable to verify timestamp for libcoreclr.so
libpthread_2_31!pthread_cond_wait+0x1e2:
00007454`1e1637b2 483d00f0ffff    cmp     rax,0FFFFFFFFFFFFF000h
0:000>
```

When I am trying to display the headers, the debugger does not recognize the debug directory:
```
0:000> !dh xyz_common

File Type: DLL
FILE HEADER VALUES
     14C machine (i386)
       3 number of sections
9FC82936 time date stamp Sat Dec 12 08:31:34 2054

       0 file pointer to symbol table
       0 number of symbols
      E0 size of optional header
    2022 characteristics
            Executable
            App can handle >2gb addresses
            DLL

OPTIONAL HEADER VALUES
     10B magic #
   48.00 linker version
   76A00 size of code
     800 size of initialized data
       0 size of uninitialized data
   788D6 address of entry point
    2000 base of code
         ----- new -----
0000000010000000 image base
    2000 section alignment
     200 file alignment
       3 subsystem (Windows CUI)
    4.00 operating system version
    0.00 image version
    4.00 subsystem version
   7E000 size of image
     200 size of headers
       0 checksum
0000000000100000 size of stack reserve
0000000000001000 size of stack commit
0000000000100000 size of heap reserve
0000000000001000 size of heap commit
    8560  DLL characteristics
            High entropy VA supported
            Dynamic base
            NX compatible
            No structured exception handler
            Terminal server aware
       0 [       0] address [size] of Export Directory
   78884 [      4F] address [size] of Import Directory
   7A000 [     464] address [size] of Resource Directory
       0 [       0] address [size] of Exception Directory
       0 [       0] address [size] of Security Directory
   7C000 [       C] address [size] of Base Relocation Directory
   787C4 [      38] address [size] of Debug Directory
       0 [       0] address [size] of Description Directory
       0 [       0] address [size] of Special Directory
       0 [       0] address [size] of Thread Storage Directory
       0 [       0] address [size] of Load Configuration Directory
       0 [       0] address [size] of Bound Import Directory
    2000 [       8] address [size] of Import Address Table Directory
       0 [       0] address [size] of Delay Import Directory
    2008 [      48] address [size] of COR20 Header Directory
       0 [       0] address [size] of Reserved Directory


SECTION HEADER #1
   .text name
   768DC virtual size
    2000 virtual address
   76A00 size of raw data
     200 file pointer to raw data
       0 file pointer to relocation table
       0 file pointer to line numbers
       0 number of relocations
       0 number of line numbers
60000020 flags
         Code
         (no align specified)
         Execute Read


Debug Directories(2)
	Type       Size     Address  Pointer
Can't read debug dir

SECTION HEADER #2
   .rsrc name
     464 virtual size
   7A000 virtual address
     600 size of raw data
   76C00 file pointer to raw data
       0 file pointer to relocation table
       0 file pointer to line numbers
       0 number of relocations
       0 number of line numbers
40000040 flags
         Initialized Data
         (no align specified)
         Read Only

SECTION HEADER #3
  .reloc name
       C virtual size
   7C000 virtual address
     200 size of raw data
   77200 file pointer to raw data
       0 file pointer to relocation table
       0 file pointer to line numbers
       0 number of relocations
       0 number of line numbers
42000040 flags
         Initialized Data
         Discardable
         (no align specified)
         Read Only
```
I think this results in being unable to load the debug symbols:
```
0:000> !sym noisy ; .reload /f /s xyz_common
noisy mode - symbol prompts on
ELFBin: No build id is found for module 'System.IO.FileSystem.dll'.  Cannot match image without load anything option set
TargetComposition: Found original image for module 'System.IO.FileSystem.dll' at '\\symbolsrv.xyz.com\symbols\System.IO.FileSystem.dll\E7FD7CA5a000\System.IO.FileSystem.dll' with matching header information
ELFBin: No build id is found for module 'HtmlSanitizer.dll'.  Cannot match image without load anything option set
TargetComposition: Unable to find any original image for module 'HtmlSanitizer.dll'
SYMSRV:  BYINDEX: 0x1C
         \\symbolsrv.xyz.com\symbols*\\symbolsrv.xyz.com\symbols\xyz*http://msdl.microsoft.com/download/symbols
         HtmlSanitizer.dll
         E5F82C6212000
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\HtmlSanitizer.dll\E5F82C6212000\HtmlSanitizer.dll - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\HtmlSanitizer.dll\E5F82C6212000\HtmlSanitizer.dl_ - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\HtmlSanitizer.dll\E5F82C6212000\file.ptr - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz\HtmlSanitizer.dll\E5F82C6212000\HtmlSanitizer.dll - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz\HtmlSanitizer.dll\E5F82C6212000\HtmlSanitizer.dl_ - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz\HtmlSanitizer.dll\E5F82C6212000\file.ptr - path not found
SYMSRV:  HTTPGET: /download/symbols/HtmlSanitizer.dll/E5F82C6212000/HtmlSanitizer.dll
SYMSRV:  HttpQueryInfo: 80190194 - HTTP_STATUS_NOT_FOUND
SYMSRV:  HTTPGET: /download/symbols/HtmlSanitizer.dll/E5F82C6212000/HtmlSanitizer.dl_
SYMSRV:  HttpQueryInfo: 80190194 - HTTP_STATUS_NOT_FOUND
SYMSRV:  HTTPGET: /download/symbols/HtmlSanitizer.dll/E5F82C6212000/file.ptr
SYMSRV:  HttpQueryInfo: 80190194 - HTTP_STATUS_NOT_FOUND
SYMSRV:  RESULT: 0x80190194
DBGHELP: c:\utils\windbg\x64\HtmlSanitizer.dll - file not found
DBGHELP: c:\utils\windbg\x64\HtmlSanitizer.dll - file not found
DBGENG:  /app/HtmlSanitizer.dll - Image mapping disallowed by non-local path.
Unable to load image /app/HtmlSanitizer.dll, Win32 error 0n2
ELFBin: No build id is found for module 'OrchardCore.Features.dll'.  Cannot match image without load anything option set
TargetComposition: Unable to find any original image for module 'OrchardCore.Features.dll'
SYMSRV:  BYINDEX: 0x1E
         \\symbolsrv.xyz.com\symbols*\\symbolsrv.xyz.com\symbols\xyz*http://msdl.microsoft.com/download/symbols
         OrchardCore.Features.dll
         9D25DFDD22000
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\OrchardCore.Features.dll\9D25DFDD22000\OrchardCore.Features.dll - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\OrchardCore.Features.dll\9D25DFDD22000\OrchardCore.Features.dl_ - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\OrchardCore.Features.dll\9D25DFDD22000\file.ptr - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz\OrchardCore.Features.dll\9D25DFDD22000\OrchardCore.Features.dll - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz\OrchardCore.Features.dll\9D25DFDD22000\OrchardCore.Features.dl_ - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz\OrchardCore.Features.dll\9D25DFDD22000\file.ptr - path not found
SYMSRV:  HTTPGET: /download/symbols/OrchardCore.Features.dll/9D25DFDD22000/OrchardCore.Features.dll
SYMSRV:  HttpQueryInfo: 80190194 - HTTP_STATUS_NOT_FOUND
SYMSRV:  HTTPGET: /download/symbols/OrchardCore.Features.dll/9D25DFDD22000/OrchardCore.Features.dl_
SYMSRV:  HttpQueryInfo: 80190194 - HTTP_STATUS_NOT_FOUND
SYMSRV:  HTTPGET: /download/symbols/OrchardCore.Features.dll/9D25DFDD22000/file.ptr
SYMSRV:  HttpQueryInfo: 80190194 - HTTP_STATUS_NOT_FOUND
SYMSRV:  RESULT: 0x80190194
DBGHELP: c:\utils\windbg\x64\OrchardCore.Features.dll - file not found
DBGHELP: c:\utils\windbg\x64\OrchardCore.Features.dll - file not found
DBGENG:  /app/OrchardCore.Features.dll - Image mapping disallowed by non-local path.
Unable to load image /app/OrchardCore.Features.dll, Win32 error 0n2
ELFBin: No build id is found for module 'xyz.Resources.dll'.  Cannot match image without load anything option set
TargetComposition: Unable to find any original image for module 'xyz.Resources.dll'
SYMSRV:  BYINDEX: 0x20
         \\symbolsrv.xyz.com\symbols*\\symbolsrv.xyz.com\symbols\xyz*http://msdl.microsoft.com/download/symbols
         xyz.Resources.dll
         F98B149189a000
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz.Resources.dll\F98B149189a000\xyz.Resources.dll - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz.Resources.dll\F98B149189a000\xyz.Resources.dl_ - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz.Resources.dll\F98B149189a000\file.ptr - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz\xyz.Resources.dll\F98B149189a000\xyz.Resources.dll - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz\xyz.Resources.dll\F98B149189a000\xyz.Resources.dl_ - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz\xyz.Resources.dll\F98B149189a000\file.ptr - path not found
SYMSRV:  HTTPGET: /download/symbols/xyz.Resources.dll/F98B149189a000/xyz.Resources.dll
SYMSRV:  HttpQueryInfo: 80190194 - HTTP_STATUS_NOT_FOUND
SYMSRV:  HTTPGET: /download/symbols/xyz.Resources.dll/F98B149189a000/xyz.Resources.dl_
SYMSRV:  HttpQueryInfo: 80190194 - HTTP_STATUS_NOT_FOUND
SYMSRV:  HTTPGET: /download/symbols/xyz.Resources.dll/F98B149189a000/file.ptr
SYMSRV:  HttpQueryInfo: 80190194 - HTTP_STATUS_NOT_FOUND
SYMSRV:  RESULT: 0x80190194
DBGHELP: c:\utils\windbg\x64\xyz.Resources.dll - file not found
DBGHELP: c:\utils\windbg\x64\xyz.Resources.dll - file not found
DBGENG:  /app/xyz.Resources.dll - Image mapping disallowed by non-local path.
Unable to load image /app/xyz.Resources.dll, Win32 error 0n2
ELFBin: No build id is found for module 'xyz.WidgetInfrastructure.dll'.  Cannot match image without load anything option set
TargetComposition: Unable to find any original image for module 'xyz.WidgetInfrastructure.dll'
SYMSRV:  BYINDEX: 0x22
         \\symbolsrv.xyz.com\symbols*\\symbolsrv.xyz.com\symbols\xyz*http://msdl.microsoft.com/download/symbols
         xyz.WidgetInfrastructure.dll
         87CD12673a4000
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz.WidgetInfrastructure.dll\87CD12673a4000\xyz.WidgetInfrastructure.dll - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz.WidgetInfrastructure.dll\87CD12673a4000\xyz.WidgetInfrastructure.dl_ - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz.WidgetInfrastructure.dll\87CD12673a4000\file.ptr - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz\xyz.WidgetInfrastructure.dll\87CD12673a4000\xyz.WidgetInfrastructure.dll - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz\xyz.WidgetInfrastructure.dll\87CD12673a4000\xyz.WidgetInfrastructure.dl_ - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz\xyz.WidgetInfrastructure.dll\87CD12673a4000\file.ptr - path not found
SYMSRV:  HTTPGET: /download/symbols/xyz.WidgetInfrastructure.dll/87CD12673a4000/xyz.WidgetInfrastructure.dll
SYMSRV:  HttpQueryInfo: 80190194 - HTTP_STATUS_NOT_FOUND
SYMSRV:  HTTPGET: /download/symbols/xyz.WidgetInfrastructure.dll/87CD12673a4000/xyz.WidgetInfrastructure.dl_
SYMSRV:  HttpQueryInfo: 80190194 - HTTP_STATUS_NOT_FOUND
SYMSRV:  HTTPGET: /download/symbols/xyz.WidgetInfrastructure.dll/87CD12673a4000/file.ptr
SYMSRV:  HttpQueryInfo: 80190194 - HTTP_STATUS_NOT_FOUND
SYMSRV:  RESULT: 0x80190194
DBGHELP: c:\utils\windbg\x64\xyz.WidgetInfrastructure.dll - file not found
DBGHELP: c:\utils\windbg\x64\xyz.WidgetInfrastructure.dll - file not found
DBGENG:  /app/xyz.WidgetInfrastructure.dll - Image mapping disallowed by non-local path.
Unable to load image /app/xyz.WidgetInfrastructure.dll, Win32 error 0n2
ELFBin: No build id is found for module 'OpenIddict.Core.dll'.  Cannot match image without load anything option set
TargetComposition: Unable to find any original image for module 'OpenIddict.Core.dll'
SYMSRV:  BYINDEX: 0x24
         \\symbolsrv.xyz.com\symbols*\\symbolsrv.xyz.com\symbols\xyz*http://msdl.microsoft.com/download/symbols
         OpenIddict.Core.dll
         FFC8D9C558000
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\OpenIddict.Core.dll\FFC8D9C558000\OpenIddict.Core.dll - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\OpenIddict.Core.dll\FFC8D9C558000\OpenIddict.Core.dl_ - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\OpenIddict.Core.dll\FFC8D9C558000\file.ptr - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz\OpenIddict.Core.dll\FFC8D9C558000\OpenIddict.Core.dll - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz\OpenIddict.Core.dll\FFC8D9C558000\OpenIddict.Core.dl_ - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz\OpenIddict.Core.dll\FFC8D9C558000\file.ptr - path not found
SYMSRV:  HTTPGET: /download/symbols/OpenIddict.Core.dll/FFC8D9C558000/OpenIddict.Core.dll
SYMSRV:  HttpQueryInfo: 80190194 - HTTP_STATUS_NOT_FOUND
SYMSRV:  HTTPGET: /download/symbols/OpenIddict.Core.dll/FFC8D9C558000/OpenIddict.Core.dl_
SYMSRV:  HttpQueryInfo: 80190194 - HTTP_STATUS_NOT_FOUND
SYMSRV:  HTTPGET: /download/symbols/OpenIddict.Core.dll/FFC8D9C558000/file.ptr
SYMSRV:  HttpQueryInfo: 80190194 - HTTP_STATUS_NOT_FOUND
SYMSRV:  RESULT: 0x80190194
DBGHELP: c:\utils\windbg\x64\OpenIddict.Core.dll - file not found
DBGHELP: c:\utils\windbg\x64\OpenIddict.Core.dll - file not found
DBGENG:  /app/OpenIddict.Core.dll - Image mapping disallowed by non-local path.
Unable to load image /app/OpenIddict.Core.dll, Win32 error 0n2
ELFBin: No build id is found for module 'xyz.Cms.Web.dll'.  Cannot match image without load anything option set
TargetComposition: Unable to find any original image for module 'xyz.Cms.Web.dll'
SYMSRV:  BYINDEX: 0x26
         \\symbolsrv.xyz.com\symbols*\\symbolsrv.xyz.com\symbols\xyz*http://msdl.microsoft.com/download/symbols
         xyz.Cms.Web.dll
         F5A44AFD24000
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz.Cms.Web.dll\F5A44AFD24000\xyz.Cms.Web.dll - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz.Cms.Web.dll\F5A44AFD24000\xyz.Cms.Web.dl_ - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz.Cms.Web.dll\F5A44AFD24000\file.ptr - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz\xyz.Cms.Web.dll\F5A44AFD24000\xyz.Cms.Web.dll - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz\xyz.Cms.Web.dll\F5A44AFD24000\xyz.Cms.Web.dl_ - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz\xyz.Cms.Web.dll\F5A44AFD24000\file.ptr - path not found
SYMSRV:  HTTPGET: /download/symbols/xyz.Cms.Web.dll/F5A44AFD24000/xyz.Cms.Web.dll
SYMSRV:  HttpQueryInfo: 80190194 - HTTP_STATUS_NOT_FOUND
SYMSRV:  HTTPGET: /download/symbols/xyz.Cms.Web.dll/F5A44AFD24000/xyz.Cms.Web.dl_
SYMSRV:  HttpQueryInfo: 80190194 - HTTP_STATUS_NOT_FOUND
SYMSRV:  HTTPGET: /download/symbols/xyz.Cms.Web.dll/F5A44AFD24000/file.ptr
SYMSRV:  HttpQueryInfo: 80190194 - HTTP_STATUS_NOT_FOUND
SYMSRV:  RESULT: 0x80190194
DBGHELP: c:\utils\windbg\x64\xyz.Cms.Web.dll - file not found
DBGHELP: c:\utils\windbg\x64\xyz.Cms.Web.dll - file not found
DBGENG:  /app/xyz.Cms.Web.dll - Image mapping disallowed by non-local path.
Unable to load image /app/xyz.Cms.Web.dll, Win32 error 0n2
ELFBin: No build id is found for module 'System.Reflection.dll'.  Cannot match image without load anything option set
TargetComposition: Found original image for module 'System.Reflection.dll' at '\\symbolsrv.xyz.com\symbols\System.Reflection.dll\D0CBF6CF8000\System.Reflection.dll' with matching header information
ELFBin: No build id is found for module 'System.Runtime.Serialization.Json.dll'.  Cannot match image without load anything option set
TargetComposition: Found original image for module 'System.Runtime.Serialization.Json.dll' at '\\symbolsrv.xyz.com\symbols\System.Runtime.Serialization.Json.dll\CA5E4DE48000\System.Runtime.Serialization.Json.dll' with matching header information
ELFBin: No build id is found for module 'xyz.ContentLocalization.dll'.  Cannot match image without load anything option set
TargetComposition: Unable to find any original image for module 'xyz.ContentLocalization.dll'
SYMSRV:  BYINDEX: 0x2A
         \\symbolsrv.xyz.com\symbols*\\symbolsrv.xyz.com\symbols\xyz*http://msdl.microsoft.com/download/symbols
         xyz.ContentLocalization.dll
         C378E733a000
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz.ContentLocalization.dll\C378E733a000\xyz.ContentLocalization.dll - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz.ContentLocalization.dll\C378E733a000\xyz.ContentLocalization.dl_ - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz.ContentLocalization.dll\C378E733a000\file.ptr - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz\xyz.ContentLocalization.dll\C378E733a000\xyz.ContentLocalization.dll - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz\xyz.ContentLocalization.dll\C378E733a000\xyz.ContentLocalization.dl_ - path not found
SYMSRV:  UNC: \\symbolsrv.xyz.com\symbols\xyz\xyz.ContentLocalization.dll\C378E733a000\file.ptr - path not found
SYMSRV:  HTTPGET: /download/symbols/xyz.ContentLocalization.dll/C378E733a000/xyz.ContentLocalization.dll
SYMSRV:  HttpQueryInfo: 80190194 - HTTP_STATUS_NOT_FOUND
SYMSRV:  HTTPGET: /download/symbols/xyz.ContentLocalization.dll/C378E733a000/xyz.ContentLocalization.dl_
SYMSRV:  HttpQueryInfo: 80190194 - HTTP_STATUS_NOT_FOUND
SYMSRV:  HTTPGET: /download/symbols/xyz.ContentLocalization.dll/C378E733a000/file.ptr
SYMSRV:  HttpQueryInfo: 80190194 - HTTP_STATUS_NOT_FOUND
SYMSRV:  RESULT: 0x80190194
DBGHELP: c:\utils\windbg\x64\xyz.ContentLocalization.dll - file not found
DBGHELP: c:\utils\windbg\x64\xyz.ContentLocalization.dll - file not found
DBGENG:  /app/xyz.ContentLocalization.dll - Image mapping disallowed by non-local path.
Unable to load image /app/xyz.ContentLocalization.dll, Win32 error 0n2

"xyz_common" was not found in the image list.
Debugger will attempt to load "xyz_common" at given base 00000000`00000000.

Please provide the full image name, including the extension (i.e. kernel32.dll)
for more reliable results.Base address and size overrides can be given as
.reload <image.ext>=<base>,<size>.
DBGENG:  xyz_common - Partial symbol image load missing image info
DBGHELP: No header for xyz_common.  Searching for dbg file
DBGHELP: .\xyz_common.dbg - file not found
DBGHELP: xyz_common missing debug info.  Searching for pdb anyway
DBGHELP: Can't use symbol server for xyz_common.pdb - no header information available
DBGHELP: xyz_common.pdb - file not found
*** WARNING: Unable to verify timestamp for xyz_common
ELFBin: Head page of module 'xyz_common' is not mapped in core.
ELFBin: Head page of module 'xyz_common' is not mapped in core.
ELFBin: No build id is found for module 'xyz_common'.  Cannot match image without load anything option set
ELFBin: Head page of module 'xyz_common' is not mapped in core.
*** ERROR: Module load completed but symbols could not be loaded for xyz_common
DBGHELP: xyz_common_0 - no symbols loaded
Unable to add module at 00000000`00000000
```
However, I can extract the aforementioned module into a dll on disk and read its headers with dumpbin just fine, including the debug directory:
```
0:000> lmvm xyz_common
Browse full module list
start             end                 module name
00007454`16df8000 00007454`16e70000   xyz_Common   (deferred)             
    Image path: /app/xyz.Common.dll
    Image name: xyz.Common.dll
    Browse all global symbols  functions  data
    Has CLR image header, track-debug-data flag not set
    Timestamp:        ***** Invalid (9FC82936)
    CheckSum:         00000000
    ImageSize:        (header) 0007E000, (mapped) 0x00078000
    Translations:     0000.04b0 0000.04e4 0409.04b0 0409.04e4
    Information from resource tables:
0:000> ? 0007454`16e70000 - 00007454`16df8000
Evaluate expression: 491520 = 00000000`00078000
0:000> .writemem c:\temp\mydll.dll 00007454`16df8000 L?00000000`00078000
Writing 78000 bytes................................................................................................................................................................................................................................................
```
And on the Powershell command prompt:
```
C:\Temp> dir .\mydll.dll

    Directory: C:\Temp

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
-a---            9/6/2023 11:28 AM         491520 mydll.dll

C:\Temp> & "c:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.37.32822\bin\Hostx64\x86\dumpbin.exe" /headers C:\temp\mydll.dll
Microsoft (R) COFF/PE Dumper Version 14.37.32822.0
Copyright (C) Microsoft Corporation.  All rights reserved.


Dump of file C:\temp\mydll.dll

PE signature found

File Type: DLL

FILE HEADER VALUES
             14C machine (x86)
               3 number of sections
        9FC82936 time date stamp
               0 file pointer to symbol table
               0 number of symbols
              E0 size of optional header
            2022 characteristics
                   Executable
                   Application can handle large (>2GB) addresses
                   DLL

OPTIONAL HEADER VALUES
             10B magic # (PE32)
           48.00 linker version
           76A00 size of code
             800 size of initialized data
               0 size of uninitialized data
           788D6 entry point (100788D6)
            2000 base of code
           7A000 base of data
        10000000 image base (10000000 to 1007DFFF)
            2000 section alignment
             200 file alignment
            4.00 operating system version
            0.00 image version
            4.00 subsystem version
               0 Win32 version
           7E000 size of image
             200 size of headers
               0 checksum
               3 subsystem (Windows CUI)
            8560 DLL characteristics
                   High Entropy Virtual Addresses
                   Dynamic base
                   NX compatible
                   No structured exception handler
                   Terminal Server Aware
          100000 size of stack reserve
            1000 size of stack commit
          100000 size of heap reserve
            1000 size of heap commit
               0 loader flags
              10 number of directories
               0 [       0] RVA [size] of Export Directory
           78884 [      4F] RVA [size] of Import Directory
           7A000 [     464] RVA [size] of Resource Directory
               0 [       0] RVA [size] of Exception Directory
               0 [       0] RVA [size] of Certificates Directory
           7C000 [       C] RVA [size] of Base Relocation Directory
           787C4 [      38] RVA [size] of Debug Directory
               0 [       0] RVA [size] of Architecture Directory
               0 [       0] RVA [size] of Global Pointer Directory
               0 [       0] RVA [size] of Thread Storage Directory
               0 [       0] RVA [size] of Load Configuration Directory
               0 [       0] RVA [size] of Bound Import Directory
            2000 [       8] RVA [size] of Import Address Table Directory
               0 [       0] RVA [size] of Delay Import Directory
            2008 [      48] RVA [size] of COM Descriptor Directory
               0 [       0] RVA [size] of Reserved Directory


SECTION HEADER #1
   .text name
   768DC virtual size
    2000 virtual address (10002000 to 100788DB)
   76A00 size of raw data
     200 file pointer to raw data (00000200 to 00076BFF)
       0 file pointer to relocation table
       0 file pointer to line numbers
       0 number of relocations
       0 number of line numbers
60000020 flags
         Code
         Execute Read

  Debug Directories

        Time Type        Size      RVA  Pointer
    -------- ------- -------- -------- --------
    D7F96D86 cv            88 000787FC    769FC    Format: RSDS, {BD0B7AAC-1040-4E54-97CD-0767E91C1A72}, 1, D:\_wfaz\07\3\s\xyzHub\xyz\xyz.Modules\xyz.Common\obj\Automation\net6.0\xyz.Common.pdb
    00000000 repro          0 00000000        0

SECTION HEADER #2
   .rsrc name
     464 virtual size
   7A000 virtual address (1007A000 to 1007A463)
     600 size of raw data
   76C00 file pointer to raw data (00076C00 to 000771FF)
       0 file pointer to relocation table
       0 file pointer to line numbers
       0 number of relocations
       0 number of line numbers
40000040 flags
         Initialized Data
         Read Only

SECTION HEADER #3
  .reloc name
       C virtual size
   7C000 virtual address (1007C000 to 1007C00B)
     200 size of raw data
   77200 file pointer to raw data (00077200 to 000773FF)
       0 file pointer to relocation table
       0 file pointer to line numbers
       0 number of relocations
       0 number of line numbers
42000040 flags
         Initialized Data
         Discardable
         Read Only

  Summary

        2000 .reloc
        2000 .rsrc
       78000 .text
C:\Temp>
```
As you can see, the debug directory is fine.

I tried to save the dump after running `.imscan /l` (thinking it may help), but saving fails. First, the `.imgscan /l` produces a lot of worrisome messages:
```
0:000> .imgscan /l
MZ at 00007450`dbc08000, prot 00000002, type 00020000 - size 80000
...
MZ at 00007453`a3910000, prot 00000002, type 00020000 - size a1e000
  Name: System.Private.CoreLib.dll
  Loaded System.Private.CoreLib.dll module
MZ at 00007453`a3920000, prot 00000020, type 00020000 - size a1e000
MZ at 00007453`a4670000, prot 00000002, type 00020000 - size 41200
  Name: System.Threading.dll
  Loaded System.Threading.dll module
MZ at 00007453`a4680000, prot 00000020, type 00020000 - size 41200
ELFBin: No build id is found for module 'System.Threading.dll'.  Cannot match image without load anything option set
TargetComposition: Found original image for module 'System.Threading.dll' at '\\symbolsrv.xyz.com\symbols\System.Threading.dll\A5D8020C41200\System.Threading.dll' with matching header information
...
```
I am concerned with the messages like
```
ELFBin: No build id is found for module 'System.Threading.dll'.  Cannot match image without load anything option set
TargetComposition: Found original image for module 'System.Threading.dll' at '\\symbolsrv.xyz.com\symbols\System.Threading.dll\A5D8020C41200\System.Threading.dll' with matching header information
```
There are many messages like that.

Anyway, once `.imgscan /l` is done, saving the dump fails:
```
0:000> .dump /ma c:\temp\1.dmp
Creating c:\temp\1.dmp - mini user dump
GenAllocateProcessObject.GetCpuPowerInfo() failed, 0x8000ffff
GenAllocateThreadObject.GetTebInfo(0xd) failed, 0x80004002
...
GenAllocateThreadObject.GetTebInfo(0x90) failed, 0x80004002
ELFBin: Extracted build id for module 'centos6-librdkafka.so' = '031c2c86f89fd59378d817baf3fc2d9d812a52d5' to match original image
ELFBin: Unable to find any original image for module 'centos6-librdkafka.so'.  Symbols and stack unwind may be unavailable
ELFBin: No build id is found for module 'System.Configuration.ConfigurationManager.dll'.  Cannot match image without load anything option set
TargetComposition: Found original image for module 'System.Configuration.ConfigurationManager.dll' at '\\symbolsrv.xyz.com\symbols\System.Configuration.ConfigurationManager.dll\FCAAB05F62000\System.Configuration.ConfigurationManager.dll' with matching header information
ELFBin: No build id is found for module 'Microsoft.AI.DependencyCollector.dll'.  Cannot match image without load anything option set
TargetComposition: Found original image for module 'Microsoft.AI.DependencyCollector.dll' at '\\symbolsrv.xyz.com\symbols\Microsoft.AI.DependencyCollector.dll\A58F3C5024000\Microsoft.AI.DependencyCollector.dll' with matching header information
ELFBin: Extracted build id for module 'libSystem.IO.Compression.Native.so' = '64d083b32230f9eb64e66921ad24c40484db3470' to match original image
ELFBin: Found original image for module 'libSystem.IO.Compression.Native.so' at '\\symbolsrv.xyz.com\symbols\libSystem.IO.Compression.Native.so\elf-buildid-64d083b32230f9eb64e66921ad24c40484db3470\libSystem.IO.Compression.Native.so' with matching build id
ELFBin: Extracted build id for module 'libssl.so.1.1' = '73051be38172c9aae9c8bbb133f497ea34c246f2' to match original image
ELFBin: Unable to find any original image for module 'libssl.so.1.1'.  Symbols and stack unwind may be unavailable
ELFBin: Extracted build id for module 'libz.so.1.2.11' = '6d245aa7fed087c98525c2e9d3cf4d3d09addf5c' to match original image
ELFBin: Unable to find any original image for module 'libz.so.1.2.11'.  Symbols and stack unwind may be unavailable
ELFBin: Extracted build id for module 'libicui18n.so.67.1' = '6bc1c815422672b74fde1b1c0803e4c9effae2a7' to match original image
ELFBin: Unable to find any original image for module 'libicui18n.so.67.1'.  Symbols and stack unwind may be unavailable
ELFBin: Extracted build id for module 'libicuuc.so.67.1' = 'f871bbd529a02abd860f0d16b842b5b20234cb49' to match original image
ELFBin: Unable to find any original image for module 'libicuuc.so.67.1'.  Symbols and stack unwind may be unavailable
ELFBin: Extracted build id for module 'libSystem.Native.so' = '11379be8bade431867c503592c771bfe6f1f09f8' to match original image
ELFBin: Found original image for module 'libSystem.Native.so' at '\\symbolsrv.xyz.com\symbols\libSystem.Native.so\elf-buildid-11379be8bade431867c503592c771bfe6f1f09f8\libSystem.Native.so' with matching build id
ELFBin: Extracted build id for module 'libappdprofiler_glibc.so' = 'ffe5e19e6071e46d8276d18d0691b1f0cba47955' to match original image
ELFBin: Unable to find any original image for module 'libappdprofiler_glibc.so'.  Symbols and stack unwind may be unavailable
ELFBin: Extracted build id for module 'libappdprofiler.so' = '40493a9735046fba52243f3922c238fe6e0800e7' to match original image
ELFBin: Unable to find any original image for module 'libappdprofiler.so'.  Symbols and stack unwind may be unavailable
GenGetProcessInfo.EnumFunctionTables(0xd) failed, 0x80004002
QuerySystemMemoryInformation failed, 0x80004001
QueryProcessVmCounters failed, 0x80004001
WriteMemoryInfo.QueryVirtual(0x7ffff15f6000) failed, 0x8000000b
Dump creation failed, HRESULT 0x8000000B
    "The operation attempted to access data outside the valid range"
```

**Comment by KenSykes** [comment]
We are cleaning up the issues list. This bug is older than six months. If you still see this issue in the current debugger and would like to see a fix please submit a new report.\ Thanks!
